examples/typedfields/app/page.tsx

```diff
@@ -1,37 +1,73 @@
-import type { TypedFieldsExamplePageFields } from '@/gql/graphql';
-import { PAGE_QUERY_BY_HANDLE } from '@/queries/pages';
 import nacelleClient from '@/services/nacelleClient';
 import ArticleLinks from '@/components/ArticleLinks';
+import { FragmentType, getFragmentData, graphql } from '@/gql';
+
+export const PAGE_FIELDS_FRAGMENT = graphql(/* GraphQL */ `
+  fragment PageFields_PageFields on TypedFieldsExamplePageFields {
+    __typename
+    handle
+    title
+    sections {
+      edges {
+        node {
+          __typename
+          ... on Node {
+            nacelleEntryId
+          }
+          ...Article_Content
+          ...Links_Content
+        }
+      }
+    }
+  }
+`);
+
+export const PAGE_QUERY_BY_HANDLE = graphql(/* GraphQL */ `
+  query PageByHandle($handle: String!) {
+    allContent(filter: { type: "page", handles: [$handle] }) {
+      edges {
+        node {
+          handle
+          nacelleEntryId
+          typedFields {
+            __typename
+            ...PageFields_PageFields
+          }
+        }
+      }
+    }
+  }
+`);
 
 export default async function HomePage(): Promise<JSX.Element> {
-  const data = await getData();
+  const data = await getData().then((pageData) =>
+    getFragmentData(PAGE_FIELDS_FRAGMENT, pageData)
+  );
 
   if (!data) return <div>404</div>;
 
-  const sections = data?.sections?.edges?.map((edge, i) => {
-    const { nacelleEntryId, typedFields } = edge?.node || {};
-
-    switch (typedFields?.__typename) {
-      case 'TypedFieldsExampleLinksFields':
-        return <ArticleLinks key={nacelleEntryId} content={typedFields} />;
-      default:
-        return (
-          <div key={nacelleEntryId}>
-            Component not found for type {typedFields?.__typename}
-          </div>
-        );
-    }
-  });
+  const sections = data.sections?.edges
+    .map((edge) => edge.node)
+    .map((link) => {
+      if (link?.__typename !== 'TypedFieldsExampleLinks') return null;
+
+      return <ArticleLinks key={link.nacelleEntryId} content={link} />;
```

Comment:
I removed the `switch-case` from this file because we are only dealing with a single component.